File: 02-basic-challenges-2/02-array-intersection/array-intersection.rb
def array_intersection(arr1, arr2)
  unique_arr1 = arr1.uniq
  unique_arr2 = arr2.uniq

  unique_arr2.select { |num| unique_arr1.include?(num) }
end

intersection = array_intersection([1, 2, 3, 4, 5], [3, 4, 5, 6, 7])
puts intersection.inspect

File: 02-basic-challenges-2/02-array-intersection/array-intersection.js
// returns array of common elements that appear in both arrays
function arrayIntersection(arr1, arr2) {
  const uniqueArr1 = [...new Set(arr1)];
  const uniqueArr2 = [...new Set(arr2)];

  // return uniqueArr1.reduce((result, num) => {
  //   uniqueArr2.includes(num) && result.push(num);
  //   return result;
  // }, []);

  return uniqueArr1.filter((el) => uniqueArr2.includes(el));
}

module.exports = arrayIntersection;
